traitlets/traitlets.py

```diff
@@ -144,6 +144,24 @@ def parse_notifier_name(name):
             assert isinstance(n, string_types), "names must be strings"
         return name
 
+def _register_callback(log, handler, name):
+    if name not in log:
+        nlist = []
+        log[name] = nlist
+    else:
+        nlist = log[name]
+    if handler not in nlist:
+        nlist.append(handler)
+
+def _unregister_callback(log, handler, name):
+    if name in log:
+        try:
+            if handler is None:
+                del log[name]
+            else:
+                log[name].remove(handler)
+        except ValueError:
+            pass
```

Comment:
Why moving the content of `_add_notifiers` and `_remove_notifiers` to `_register_callback` and `_unregister_callback`?